packages/functions/src/dataTriggers.ts

```diff
@@ -440,3 +440,54 @@ export const createAttendedSlotOnAttendance = functions
     );
     await Promise.all(updates);
   });
+
+// on write of bookings coll
+// check if update change.after.exists && contains customer data not booked slots
+// update customer doc in customers collection
+
+/**
+ * Data trigger used to update attendance entries for slot whenever customer books a certain slot + interval.
+ *
+ * - listens to `organizations/{organization}/bookings/{secretKey}/bookedSlots/{slotId}`
+ * - writes to `organizations/{organization}/attendnace/{slotId}` - updates entry for `attendances[customerId]` leaving the rest of the doc unchanged
+ */
+export const updateCustomerWithDataFromBookings = functions
```

Comment:
This trigger should not be needed. Athletes editing a form will hit a cloud function on submit. The cloud function will check that they provided a valid secret key, and if they did it will update the `customers` collection.
The (poorly named, my bad!) `addIdAndSecretKey` method will take care of copying the new data to the `bookings` collection.
This is to make sure we have one authoritative, primary place to hold the data (the `customers` collection) and the other spot (the `bookings` collection) is treated as secondary, and can only get its values from the primary source.

I also expected this to create a "trigger loop": `updateCustomerWithDataFromBookings` will trigger when there's an update to `bookings`, and it will update `customers` accordingly. This in turn will cause the `addIdAndSecretKey` trigger to run; it will update the `bookings` collection accordingly. This in turn will trigger `updateCustomerWithDataFromBookings`, rinse and repeat...
The above is not happening and I'm not sure why. Still, I firmly believe in the necesssity to have one place as primary and another one as secondary, and only mutate the primary one via client code. The secondary one will be updated only by triggers.

Reply:
Also, I believe the function docs were not updated after copy pasting. No need to fix them, since this trigger should go.